parseAPI/src/debug_parse.C

```diff
@@ -59,23 +57,36 @@ int Dyninst::ParseAPI::parsing_printf_int(const char *format, ...)
 
     if(!dyn_debug_parsing) return 0;
     if(NULL == format) return -1;
-    if (log_file == NULL) {
-        char filename[128];
-#if defined(_OPENMP)
-        snprintf(filename, 128, "%s-%d.txt", getenv("DYNINST_DEBUG_PARSING"), omp_get_thread_num());
-#else
-        snprintf(filename, 128, "%s-%d.txt", getenv("DYNINST_DEBUG_PARSING"), 0);
+
+    auto const id = []() -> int {
+#ifdef _OPENMP
+      return omp_get_thread_num();
 #endif
+      return 0;
+    }();
 
-        log_file = fopen(filename, "w");
-    }
+    auto msg = "[thread " + std::to_string(id) + "] " + std::string(format);
 
     va_list va;
     va_start(va,format);
-    int ret = vfprintf(log_file, format, va);
-    fflush(log_file);
-    va_end(va);
 
+    auto ret = [&]() {
+      int v{};
+      #pragma omp critical
+      {
+#ifdef __clang__
```

Comment:
To preserve the -Wformat-nonliteral, which is a useful security property, I'd suggest removing the creation of a new format string, by just using `fprintf(stderr, "[thread %u] ", id);` and then the `vfprintf(stderr, format, va);` without the `#pragma`'s.

Reply:
Not sure why I didn't think of that... Changed.